File: javascript/bob/bob.js
var Bob = function() {};

Bob.prototype.hey = function(input) {
	var result
	if (input == input.toUpperCase() && input != input.toLowerCase()) {
		result = 'Whoa, chill out!'	
	} else if (input[input.length - 1] == '?'){
		result = 'Sure.'
	} else if (input.trim() == "" ){
		result = 'Fine. Be that way!'
	} else {
		result = 'Whatever.'
	}
	 	 	
 	return result 
		
};

module.exports = Bob;

File: python/bob/bob.py
def response(hey_bob):
    answer = 'Whatever.'

    cleaned_hey_bob = hey_bob.strip()

    if len(cleaned_hey_bob) == 0:
        answer = 'Fine. Be that way!'

    if cleaned_hey_bob.isupper():
        if cleaned_hey_bob.endswith('?'):
            answer = "Calm down, I know what I'm doing!"
        else:
            answer = "Whoa, chill out!"

    if cleaned_hey_bob.endswith('?') and not cleaned_hey_bob.isupper():
        answer = "Sure."

    return answer
